File: Leetcode/1512. Number of Good Pairs/solution.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        mem = {}
        pairs = 0
        for num in nums:
            count = mem.get(num, 0)

            if count:
                pairs += count

            mem[num] = count + 1

        return pairs

File: Leetcode/1512. Number of Good Pairs/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn num_identical_pairs(nums: Vec<i32>) -> i32 {
        let mut seen = HashMap::new();
        let mut pairs = 0;
        for num in nums {
            let mut count = seen.get(&num);
            
            let freq = match count {
                Some(num) => *num,
                None => 0,
            };
            
            if freq > 0 {
                pairs += freq;
            }
            seen.insert(num, freq + 1);
        }
        pairs
    }
}
